### PR TITLE
odin 0.12.0 (new formula)

### PR DESCRIPTION
**Pull request by vladimyr**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/odin.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class Odin < Formula
+  desc "The Odin Programming Language"
+  homepage "https://odin-lang.org/"
+  url "https://github.com/odin-lang/Odin/archive/v0.12.0.tar.gz"
+  sha256 "8356124c8cc7e08ac39872e5bb10593a412e67f81df621124097facd9b2b26cc"
+
+  depends_on "llvm"
```
Comment (SMillerDev) [inline Formula/odin.rb:7]: Does it not work with macOS llvm?

File: Formula/odin.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  uses_from_macos "iconv"
+
+  def install
+    system "make", "release", "CC=#{ENV.cc}"
```
Comment (SMillerDev) [inline Formula/odin.rb:12]: CC should already be set by homebrew

File: Formula/odin.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    libexec.install "odin", "core", "shared"
+    (bin/"odin").write <<~EOS
+      #!/bin/sh
+      export PATH="/usr/local/opt/llvm/bin:$PATH"
```
Comment (SMillerDev) [inline Formula/odin.rb:16]: Please use `#{Formula["llvm"].opt_bin}` instead of the `/usr/local` path.

File: Formula/odin.rb (hunk 1 above)
Comment (vladimyr) [reply Formula/odin.rb:7]: It works but it also needs other binaries from llvm toolchain not present in default installation like `opt`.

**Comment by SMillerDev** [comment]
Please squash all your commits into one.

**Comment by vladimyr** [comment]
> Please squash all your commits into one.

Done ✅

File: Formula/odin.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  uses_from_macos "iconv"
+
+  def install
+    system "make", "release"
+    libexec.install "odin", "core", "shared"
+    (bin/"odin").write <<~EOS
+      #!/bin/sh
+      export PATH="#{Formula["llvm"].opt_bin}:$PATH"
+      #{libexec}/odin "$@"
+    EOS
```
Comment (bayandin) [inline Formula/odin.rb:19]:
```suggestion
    (bin/"odin").write_env_script(libexec/"odin", :PATH => "#{Formula["llvm"].opt_bin}:$PATH")
```

Comment (vladimyr) [reply Formula/odin.rb]: Thanks for the suggestion but at the same time I realized I should use `exec -a` to adjust Odin's `argv0`.

File: Formula/odin.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+class Odin < Formula
+  desc "The Odin Programming Language"
+  homepage "https://odin-lang.org/"
+  url "https://github.com/odin-lang/Odin/archive/v0.12.0.tar.gz"
+  sha256 "8356124c8cc7e08ac39872e5bb10593a412e67f81df621124097facd9b2b26cc"
+  head "https://github.com/odin-lang/Odin.git"
+
+  depends_on "llvm"
+
+  uses_from_macos "iconv"
```
Comment (bayandin) [inline Formula/odin.rb:10]:
```suggestion
  uses_from_macos "libiconv"
```

We need this to fix warning:
```
==> FAILED
== /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/odin.rb ==
C: 10:  3: uses_from_macos should only be used for macOS dependencies, not iconv.
```

**Review by bayandin — CHANGES_REQUESTED** [review]

File: Formula/odin.rb (hunk 5 above)
Comment (vladimyr) [reply Formula/odin.rb:10]: Thanks for drawing my attention to it. I should've been more diligent!

**Comment by bayandin** [comment]
Thanks @vladimyr for your contribution to Homebrew! 🎉 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**